src/unity/toolkits/supervised_learning/logistic_regression_opt_interface.cpp

```diff
@@ -194,6 +194,19 @@ logistic_regression_opt_interface::get_status_header(const std::vector<std::stri
   return header;
 }
 
+variant_type logistic_regression_opt_interface::get_validation_accuracy() {
+  DASSERT_TRUE(valid_data.num_rows() > 0);
+  auto eval_results = smodel.evaluate(valid_data, "train");
+  auto results = eval_results.find("accuracy");
+  return results->second;
+}
+
+variant_type logistic_regression_opt_interface::get_training_accuracy() {
+  auto eval_results = smodel.evaluate(data, "train");
+  auto results = eval_results.find("accuracy");
+  return results->second;
```

Comment:
Are there any cases where these are not set?  If so, we should test for it.  If not, let's just put in an `ASSERT_TRUE(results != eval_results.end())` or, actually, use this form:
```
variant_type accuracy = eval_results.at("accuracy");  // .at(...) throws if not preset
return variant_get_type<flexible_type>(accuracy).to<double>();
```

 The existing code will segfault if accuracy is not present.

Reply:
No, I'm not confident in that. I thought you were. I guess I misunderstood.

I decided to add an explicit check and a `log_and_throw`, so the user gets an intelligent error message.